tests/integration/components/donations/donation-progress-test.js

```diff
@@ -0,0 +1,34 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent('donations/donation-progress', 'Integration | Component | donations/donation progress', {
+  integration: true
+});
+
+test('it renders proper information', function(assert) {
+  assert.expect(4);
+
+  let mockGoal = { amount: 1000, description: 'Lorem ipsum' };
+  this.set('donationGoal', mockGoal);
+
+  this.render(hbs`{{donations/donation-progress donationGoal=donationGoal amountDonated=500}}`);
+
+  assert.equal(this.$('.amount .value').text().trim(), '$500', 'Correct amount is rendered');
+  assert.equal(this.$('.percentage .value').text().trim(), '50%', 'Correct percentage is rendered');
+  assert.equal(this.$('.progress-bar').length, 1, 'Progress bar component is rendered');
+
+  assert.equal(this.$('.description').text().trim(), mockGoal.description, 'Goal description is rendered');
+});
+
+test('it renders decimal values if there are any', function(assert) {
+  assert.expect(3);
+
+  let mockGoal = { amount: 1000, description: 'Lorem ipsum' };
+  this.set('donationGoal', mockGoal);
+
+  this.render(hbs`{{donations/donation-progress donationGoal=donationGoal amountDonated=500.50}}`);
+
+  assert.equal(this.$('.amount .value').text().trim(), '$500.50', 'Correct amount is rendered');
+  assert.equal(this.$('.percentage .value').text().trim(), '50.05%', 'Correct percentage is rendered');
+  assert.equal(this.$('.progress-bar').length, 1, 'Progress bar component is rendered');
```

Comment:
I looked further into this and tried to create a solution that makes sense to me.

This is the documentation for using page objects in integration tests:

http://ember-cli-page-object.js.org/docs/v1.6.x/quickstart#integration-tests

As you can see, the idea is that we have a page object for the acceptance tests and then reuse it in integration tests.

Thing is, we do not have such a page object yet, and to me, it seems like, if we define a full-fledged page object for acceptance testing of the page that would contain this component, it's misleading to use it for testing the component alone.

Instead, my idea was to define `pages/components/donations/donation-progress.js` as a page object component. We then import that component into our test and create a page object using `PageObject.create(donationProgressComponent)`.

Additionally, we have to set and remove it's context before and after each test. That being said, the tests are much cleaner that way, so I like the solution. It's just that I would also love if there was a way to wrap the whole process into a helper of some sort, possibly something like

```JavaScript
moduleForComponentUsingPageObject(path, moduleName, pageObjectComponent) {
  let page = PageObject.create(pageObjectComponent);
  moduleForComponent(path, moduleName, {
    integration: true,
    beforeEach: function() { page.setContext(this); },
    afterEach: function() { page.removeContext(this) }
  })

  return page;
}
```

Note that we would have to return the page, since we then need to access it in our tests.

We would use this helper along the lines of 

```Javascript
import moduleForComponentUsingPageObject from '../../../helpers/module-for-component-using-page-object';
import donationProgressComponent from '../../../pages/components/donations/donation-progress';

let page = moduleForComponentUsingPageObject('donations/donation-progress', 'Integration | Component | donations/donation progress', donationProgressComponent);
```

Compared to the current 

```Javascript
import PageObject from 'ember-cli-page-object';
import donationProgressComponent from '../../../pages/components/donations/donation-progress';

let page = PageObject.create(donationProgressComponent);

moduleForComponent('donations/donation-progress', 'Integration | Component | donations/donation progress', {
  integration: true,
  beforeEach() {
    page.setContext(this);
  },

  afterEach() {
    page.removeContext();
  }
});
```

I'm not sure it's that much cleaner, but it slightly cleaner, and it definitely is shorter (3 lines, compared to 13). 

That being said, there would be additional complications with such a helper, if we want to use custom before and after contexts.